gcloud-java-dns/src/main/java/com/google/gcloud/dns/ProjectInfo.java

```diff
@@ -0,0 +1,285 @@
+      return rrsetAdditionsPerChange;
+    }
+
+    /**
+     * Returns the maximum allowed number of {@link DnsRecord}s to delete per {@code
+     * ChangesRequest}.
+     */
+    public Integer rrsetDeletionsPerChange() {
+      return rrsetDeletionsPerChange;
+    }
+
+    /**
+     * Returns the maximum allowed number of {@link DnsRecord}s per {@code ManagedZone} in the
+     * project.
+     */
+    public Integer rrsetsPerManagedZone() {
+      return rrsetsPerManagedZone;
+    }
+
+    /**
+     * Returns the maximum allowed size for total records in one ChangesRequest in bytes.
+     */
+    public Integer totalRrdataSizePerChange() {
+      return totalRrdataSizePerChange;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      return (other instanceof Quota) && this.toPb().equals(((Quota) other).toPb());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(zones, resourceRecordsPerRrset, rrsetAdditionsPerChange,
+          rrsetDeletionsPerChange, rrsetsPerManagedZone, totalRrdataSizePerChange);
+    }
+
+    com.google.api.services.dns.model.Quota toPb() {
+      com.google.api.services.dns.model.Quota pb = new com.google.api.services.dns.model.Quota();
+      pb.setManagedZones(zones);
+      pb.setResourceRecordsPerRrset(resourceRecordsPerRrset);
+      pb.setRrsetAdditionsPerChange(rrsetAdditionsPerChange);
+      pb.setRrsetDeletionsPerChange(rrsetDeletionsPerChange);
+      pb.setRrsetsPerManagedZone(rrsetsPerManagedZone);
+      pb.setTotalRrdataSizePerChange(totalRrdataSizePerChange);
+      return pb;
+    }
+
+    static Quota fromPb(com.google.api.services.dns.model.Quota pb) {
+      Quota quota = new Quota(pb.getManagedZones(),
+          pb.getResourceRecordsPerRrset(),
+          pb.getRrsetAdditionsPerChange(),
+          pb.getRrsetDeletionsPerChange(),
+          pb.getRrsetsPerManagedZone(),
+          pb.getTotalRrdataSizePerChange()
+      );
+      return quota;
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("zones", zones)
+          .add("resourceRecordsPerRrset", resourceRecordsPerRrset)
+          .add("rrsetAdditionsPerChange", rrsetAdditionsPerChange)
+          .add("rrsetDeletionsPerChange", rrsetDeletionsPerChange)
+          .add("rrsetsPerManagedZone", rrsetsPerManagedZone)
+          .add("totalRrdataSizePerChange", totalRrdataSizePerChange)
+          .toString();
+    }
+  }
+
+  /**
+   * A builder for {@code ProjectInfo}.
+   */
+  static class Builder {
+    private String id;
+    private BigInteger number;
+    private Quota quota;
+
+    private Builder() {
+    }
+
+    /**
+     * Sets an id of the project.
+     */
+    Builder id(String id) {
+      this.id = checkNotNull(id);
+      return this;
+    }
+
+    /**
+     * Sets a number of the project.
+     */
+    Builder number(BigInteger number) {
+      this.number = checkNotNull(number);
```

Comment:
is that the project number?

If so (and for the above), I find it strange to have this information specific to DNS.

Project id would be available from the options, though maybe we should have part of the common Service api
a way to get the id (and number), which could be null for some services.